x-pack/plugins/watcher/common/types/watch_types.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface ExecutedWatchResults {
+  id: string;
+  watchId: string;
+  details: any;
+  startTime: Date;
+  watchStatus: {
+    state: string;
+    actionStatuses: [
```

Comment:
I think this means that actionStatuses can only be an array with one member of this shape.  I think what you want here is:
```
Array<{ state: string; lastExecutionReason: string; }>